### PR TITLE
Category info edit

### PR DESCRIPTION
**Pull request by Gigiman**
Editing category info functionality added

### CODE REVIEW

File: front/client/components/CategoryInfoEdit.vue [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    if (status === 'CategoryStub') {
+      return 'Stub'
+    } else if (status === 'CategoryWIP') {
+      return 'Work in progress'
+    } else if (status === 'CategoryFinished') {
+      return 'Complete'
+    }
+  }
+
+  setCategoryStatusRaw () {
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue]: 1) This function doesnt set anything, it returns value. So name is really confusing.
2) Second, have you heard about switch? 
3) There are CategoryStatus enum, check it and replace raw strings to its properties.

File: front/client/components/CategoryInfoEdit.vue [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  onEcosysemChange () {
+    this.onCheckboxChange(this.ecosystem, 'ItemEcosystemSection')
+  }
+
+  @Watch('notes')
+  onNotesChange () {
+    this.onCheckboxChange(this.notes, 'ItemNotesSection')
+  }
+
+  transformCategoryStatus (status: string) {
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue:121]: Next function is almost the same.

File: front/client/components/CategoryInfoEdit.vue [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+  itemValidationRules: Function[] = [
+    (x: string) => !!x || 'Item name can not be empty'
+  ]
+
+  beforeMount () {
+    this.$vuetify.icons.checkboxOff = this.$vuetify.icons.square
+  }
+
+  onCheckboxChange(checkboxType: boolean, checkboxName: string) {
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue]: This function is hard to understand. Refactor it. You can use array as v-model for all checkboxes so you dont need all this watchers and this function too. Check out vue doc, or event vuetify doc (there is such example with vuetify checkboxes)
Also, there are native array functions which do the same as "_includes" and "_remove", so you dont need to import excess code from lodash.

File: front/client/components/CategoryInfoEdit.vue [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  isValid: boolean = false
+
+  items = ['Complete', 'Work in progress', 'Stub']
+
+  itemValidationRules: Function[] = [
+    (x: string) => !!x || 'Item name can not be empty'
+  ]
+
+  beforeMount () {
+    this.$vuetify.icons.checkboxOff = this.$vuetify.icons.square
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue:109]: Wtf is this? Why you replace one icon by another? And what happens if I want to use "checkboxOff" icon somewhere after this?

File: front/client/components/CategoryInfoEdit.vue [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+          flat
+          color="primary"
+          @click.native="close"
+        >
+          Cancel
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template> -->
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue]: Some strange arrrow down here.

File: front/client/page/SearchResults.vue [hunk 6]
```diff
@@ -17,7 +17,7 @@
             »
           <a-link
             openInNewTab
```
Comment (avele) [inline front/client/page/SearchResults.vue:19]: I think its better to remove all this chages and replace links to router-link pointing to category component in another pull-request

File: front/client/components/CategoryInfoEdit.vue [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+  prosCons: boolean = true
+  ecosystem: boolean = true
+  notes: boolean = true
+  sections: any[] = []
+  isValid: boolean = false
+
+  items = ['Complete', 'Work in progress', 'Stub']
+
+  itemValidationRules: Function[] = [
+    (x: string) => !!x || 'Item name can not be empty'
```
Comment (avele) [inline front/client/components/CategoryInfoEdit.vue]: You call both 'Title' and 'Group'. Make accurate validation rules for each input (you can even do it in markup)

File: front/client/components/CategoryInfoEdit.vue (hunk 1 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue]: done

File: front/client/components/CategoryInfoEdit.vue (hunk 4 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue:109]: this is the problem with the svg icons we were discussing, some icons cant be replaced with the svg icon. Icon replacement is a hack from vuetify issue

File: front/client/components/CategoryInfoEdit.vue (hunk 5 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue]: done

File: front/client/components/CategoryInfoEdit.vue (hunk 3 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue]: done

File: front/client/components/CategoryInfoEdit.vue (hunk 7 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue]: done

File: front/client/components/CategoryInfoEdit.vue (hunk 2 above)
Comment (Gigiman) [reply front/client/components/CategoryInfoEdit.vue:121]: refactored

File: front/client/page/SearchResults.vue (hunk 6 above)
Comment (Gigiman) [reply front/client/page/SearchResults.vue:19]: added to task list in notion

**Pull request merged**